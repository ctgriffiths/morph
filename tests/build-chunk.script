#!/bin/sh
#
# Test building a chunk.
#
# Copyright (C) 2011  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

set -e

cache="$DATADIR/build-chunk-cache"
./morph --no-default-configs build repo farrokh hello.morph \
    --git-base-url="file://$DATADIR" \
    --cachedir="$cache" --keep-path
for chunk in "$cache/"*.chunk.*
do
    echo "$chunk:" | sed 's/[^.]*//'
    tar -tf "$chunk" | LC_ALL=C sort | sed '/^\.\/./s:^\./::'
    echo
done
