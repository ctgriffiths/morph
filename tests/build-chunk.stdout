.chunk.hello:
./
baserock/
baserock/hello.meta
bin/
bin/hello

